plugins/modules/dw_virtual_warehouse.py

```diff
@@ -0,0 +1,433 @@
+                    # End Config check
+            else:
+                self.module.fail_json(msg="State %s is not valid for this module" % self.state)
+            # End handling Virtual Warehouse exists
+        else:
+            # Begin handling Virtual Warehouse not found
+            if self.state == 'absent':
+                self.module.warn(
+                    "DW Virtual Warehouse %s already absent in Cluster %s" % (self.name, self.cluster_id))
+            elif self.state == 'present':
+                if self.module.check_mode:
+                    pass
+                else:
+                    vw_id = self.cdpy.dw.create_vw(cluster_id=self.cluster_id,
+                                                   dbc_id=self.dbc_id, vw_type=self.type, name=self.name,
+                                                   template=self.template,
+                                                   autoscaling_min_cluster=self.autoscaling_min_nodes,
+                                                   autoscaling_max_cluster=self.autoscaling_max_nodes,
+                                                   common_configs=self.common_configs,
+                                                   application_configs=self.application_configs,
+                                                   ldap_groups=self.ldap_groups, enable_sso=self.enable_sso,
+                                                   tags=self.tags)
+                    if self.wait:
+                        self.target = self.cdpy.sdk.wait_for_state(
+                            describe_func=self.cdpy.dw.describe_vw,
+                            params=dict(cluster_id=self.cluster_id, vw_id=vw_id),
+                            state=self.cdpy.sdk.STARTED_STATES, delay=self.delay, timeout=self.timeout
+                        )
+                    else:
+                        self.target = self.cdpy.dw.describe_vw(cluster_id=self.cluster_id, vw_id=vw_id)
+                    self.virtual_warehouses.append(self.target)
+            else:
+                self.module.fail_json(msg="State %s is not valid for this module" % self.state)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=CdpModule.argument_spec(
+            id=dict(required=False, type='str', default=None),
+            cluster_id=dict(required=True, type='str'),
+            dbc_id=dict(required=False, type='str', default=None),
+            type = dict(required=False, type='str', default=None),
+            name = dict(required=False, type='str', default=None),
+            template=dict(required=False, type='str', default=None),
+            autoscaling_min_nodes=dict(required=False, type='int', default=None),
+            autoscaling_max_nodes=dict(required=False, type='int', default=None),
+            common_configs=dict(required=False, type='dict', default=None),
+            application_configs=dict(required=False, type='dict', default=None),
```

Comment:
Both `common_configs` and `application_configs` need to be expanded to match documentation rules above.